ocs_sample_library_preview/SharedStreams.py

```diff
@@ -0,0 +1,483 @@
+
+    def getStreamType(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str) -> SdsType:
+        """
+        Retrieves a stream type specified by 'stream_id' in a community specified by 'community_id' from the Sds Service
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :return: the stream type as an SdsType
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.getCommunityRequestHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                stream_id=self.__base_client.encode(stream_id)) + '/Type',
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, f'Failed to get SdsStream type, {stream_id}.')
+
+        result = SdsType.fromJson(response.json())
+        return result
+
+    def getStreams(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, query: str = '', skip: int = 0,
+                   count: int = 100) -> list[SdsStream]:
+        """
+        Retrieves a list of streams associated with 'namespace_id' under the specified tenant and community
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :param query: filtering query
+        :param skip: number of streams to skip for paging
+        :param count: number of streams to limit to
+        :return: array of SdsStreams
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.getCommunityRequestHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__streams_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id),
+            params={'query': query, 'skip': skip, 'count': count},
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, 'Failed to get all SdsStreams.')
+        
+        content = response.json()
+        results: list[SdsStream] = []
+        for item in content:
+            results.append(SdsStream.fromJson(item))
+        return results
+
+    # The following section provides functionality to interact with Data
+    #  We assume the value(s) passed follow the Sds object patterns
+    #  supporting fromJson and toJson method
+
+    def getValue(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, index: int, value_class: type = None) -> Any:
```

Comment:
index is not necessarily an int

Reply:
though I see this throughout, though we seem to generally have indicies as str -which i think work for the case of number indicies though someone might have to make a tpye conversion in python to pass into the functoin

Reply:
you're right, to accommodate for timestamps etc we should probably use str here instead. I could change all of them to str here if we think that would be better, I am not sure if there was a reason for using int in the first place

Reply:
I tried out a request where I passed it a string representing a timestamp, and python didn't complain. I'm not actually sure what these type constraints even do now

Reply:
tdil, they do nothing, they are just for syntax highlighting pretty much

From towardsdatascience:
_Erroneous type annotations will do nothing more than highlight the incorrect annotation in our code editor — no errors are ever raised due to annotations.
So, if type annotations are not enforced, why use them?
Well, as we touched upon already — declaring types makes our code more explicit, and if done well, easier to read — both for ourselves and others._

So with that I guess it's really about making it clearer for users, which I guess would be str, or maybe even Any?